#
# Copyright (c) 2015, 2016, Oracle and/or its affiliates. All rights reserved.
#
avoid potential buffer overrun in enumEitherTrie
http://bugs.icu-project.org/trac/ticket/12387
diff --git a/source/common/utrie2.cpp b/source/common/utrie2.cpp
--- a/icu/source/common/utrie2.cpp
+++ b/icu/source/common/utrie2.cpp
@@ -32,6 +32,7 @@
 #include "unicode/utf.h"
 #include "unicode/utf8.h"
 #include "unicode/utf16.h"
+#include "unicode/uchar.h"
 #include "cmemory.h"
 #include "utrie2.h"
 #include "utrie2_impl.h"
@@ -554,6 +555,10 @@ enumEitherTrie(const UTrie2 *trie,
     UChar32 c, prev, highStart;
     int32_t j, i2Block, prevI2Block, index2NullOffset, block, prevBlock, nullBlock;
 
+    if(limit>UCHAR_MAX_VALUE+1) {
+	return;
+    }
+
     if(enumRange==NULL) {
         return;
     }
